--- ./filters/karbon/pdf/SvgOutputDev.cpp.orig	2020-05-14 01:51:30.000000000 -0500
+++ ./filters/karbon/pdf/SvgOutputDev.cpp	2025-10-04 15:33:04.863597106 -0500
@@ -407,7 +407,7 @@
         return;
 
     // ignore empty strings
-    if (s->getLength() == 0)
+    if (s->size() == 0)
         return;
 
     GfxFont * font = state->getFont();
@@ -423,7 +423,7 @@
         const char * p = s->c_str();
     #endif
 #endif
-    int len = s->getLength();
+    int len = s->size();
     CharCode code;
 #ifdef HAVE_POPPLER_PRE_0_82
     Unicode *u = nullptr;
