diff -urN kdevelop-24.05.0.orig/CMakeLists.txt kdevelop-24.05.0/CMakeLists.txt
--- kdevelop-24.05.0.orig/CMakeLists.txt	2024-05-17 01:44:01.000000000 +0300
+++ kdevelop-24.05.0/CMakeLists.txt	2024-05-26 23:08:16.521123192 +0300
@@ -103,28 +103,6 @@
 )
 
 set(HAVE_KSYSGUARD FALSE)
-find_package(KSysGuard CONFIG QUIET)
-set_package_properties(KSysGuard PROPERTIES
-    PURPOSE "Framework for process listing. Required for the 'Attach to Process' feature"
-    TYPE RECOMMENDED
-)
-if(NOT KSysGuard_FOUND)
-    # try old pre-Plasma 5.19 name
-    find_package(KF5SysGuard CONFIG)
-    set_package_properties(KF5SysGuard PROPERTIES
-        PURPOSE "Framework for process listing (older name of KSysGuard). Required for the 'Attach to Process' feature"
-        TYPE RECOMMENDED
-    )
-endif()
-if(KSysGuard_FOUND OR KF5SysGuard_FOUND)
-    set(HAVE_KSYSGUARD TRUE)
-endif()
-
-find_package(KDevelop-PG-Qt 1.90.90 CONFIG)
-set_package_properties(KDevelop-PG-Qt PROPERTIES
-    PURPOSE "KDevelop parser generator library. Required for the QMake Builder/Manager plugin."
-    TYPE RECOMMENDED
-)
 
 find_package(SharedMimeInfo 1.9 REQUIRED)
 
