--- kirigami-gallery-20231029git/CMakeLists.txt.orig    2023-10-29 12:29:14.000000000 +0100
+++ kirigami-gallery-20231029git/CMakeLists.txt 2023-10-29 13:29:37.069508322 +0100
@@ -41,7 +41,7 @@
     find_package(Qt${QT_MAJOR_VERSION}Widgets ${REQUIRED_QT_VERSION} REQUIRED)
 endif()
 
-ecm_find_qmlmodule(QtGraphicalEffects 1.0)
+ecm_find_qmlmodule(Qt5Compat.GraphicalEffects 6.0)
 ecm_find_qmlmodule(org.kde.kitemmodels 1.0)
 
 add_definitions(-DQT_NO_URL_CAST_FROM_STRING)

