diff -uar projectM-2.0.1-Source.orig/Common.hpp projectM-2.0.1-Source/Common.hpp
--- projectM-2.0.1-Source.orig/Common.hpp	2009-12-06 02:42:51.000000000 +0100
+++ projectM-2.0.1-Source/Common.hpp	2017-05-26 12:52:59.475660783 +0200
@@ -55,7 +55,8 @@
 
 #ifdef LINUX
 #include <cstdlib>
-#define projectM_isnan isnan
+#include <cmath>
+#define projectM_isnan std::isnan
 
 #endif
 
diff -uar projectM-2.0.1-Source.orig/MilkdropPresetFactory/Parser.cpp projectM-2.0.1-Source/MilkdropPresetFactory/Parser.cpp
--- projectM-2.0.1-Source.orig/MilkdropPresetFactory/Parser.cpp	2009-12-06 02:42:51.000000000 +0100
+++ projectM-2.0.1-Source/MilkdropPresetFactory/Parser.cpp	2017-05-26 12:52:59.476660949 +0200
@@ -1405,7 +1405,7 @@
   PerFrameEqn * per_frame_eqn;
   GenExpr * gen_expr;
 
-  if (fs == NULL)
+  if (!fs)
     return NULL;
   if (param_string == NULL)
     return NULL;
@@ -1560,7 +1560,7 @@
 
   if (preset == NULL)
     return NULL;
-  if (fs == NULL)
+  if (!fs)
     return NULL;
 
   if ((token = parseToken(fs, name)) != tEq)
@@ -1874,7 +1874,7 @@
   /* Null argument checks */
   if (preset == NULL)
     return PROJECTM_FAILURE;
-  if (fs == NULL)
+  if (!fs)
     return PROJECTM_FAILURE;
   if (token == NULL)
     return PROJECTM_FAILURE;
@@ -2165,7 +2165,7 @@
 
   if (token == NULL)
     return PROJECTM_FAILURE;
-  if (fs == NULL)
+  if (!fs)
     return PROJECTM_FAILURE;
   if (preset == NULL)
     return PROJECTM_FAILURE;
@@ -2347,7 +2347,7 @@
   if (token == NULL)
 
     return PROJECTM_FAILURE;
-  if (fs == NULL)
+  if (!fs)
     return PROJECTM_FAILURE;
   if (preset == NULL)
     return PROJECTM_FAILURE;
