diff --git a/kcms/libkwindevices/inputdevice.h b/kcms/libkwindevices/inputdevice.h
index 8edf6ea9b702ca563f2b580b6be2645884f5da55..f3b420c12e40cfffaf57584613e26e9ac5251022 100644
--- a/kcms/libkwindevices/inputdevice.h
+++ b/kcms/libkwindevices/inputdevice.h
@@ -98,7 +98,7 @@ public:
 
     QRectF inputArea() const
     {
-        return m_inputArea.isSupported() ? m_inputArea.value() : QRectF(0, 0, 1, 1);
+        return m_inputArea.value();
     }
     void setInputArea(const QRectF &inputArea);
     bool supportsInputArea() const
