diff --git a/kcms/dateandtime/main.cpp b/kcms/dateandtime/main.cpp
index 12ef15871021c0405522111b657b57c4be1b5b04..b7f1fdee17cef13a0009ec6cb2afc7c008833b25 100644
--- a/kcms/dateandtime/main.cpp
+++ b/kcms/dateandtime/main.cpp
@@ -68,8 +68,8 @@ bool KclockModule::kauthSave()
         QDateTime newTime = dtime->userTime();
         qDebug() << "Set date to " << dtime->userTime();
         helperargs[QStringLiteral("date")] = true;
-        helperargs[QStringLiteral("newdate")] = QString::number(newTime.currentSecsSinceEpoch());
-        helperargs[QStringLiteral("olddate")] = QString::number(::time(nullptr));
+        helperargs[QStringLiteral("newdate")] = QString::number(newTime.toSecsSinceEpoch());
+        helperargs[QStringLiteral("olddate")] = QString::number(QDateTime::currentSecsSinceEpoch());
     }
 
     QString selectedTimeZone = dtime->selectedTimeZone();
@@ -81,6 +81,7 @@ bool KclockModule::kauthSave()
     }
 
     Action action(authActionName());
+    action.setHelperId(QStringLiteral("org.kde.kcontrol.kcmclock"));
     action.setArguments(helperargs);
 
     ExecuteJob *job = action.execute();
