--- ktorrent-24.01.75/CMakeLists.txt.orig	2023-11-07 21:07:03.000000000 +0100
+++ ktorrent-24.01.75/CMakeLists.txt	2023-11-12 20:36:10.795589530 +0100
@@ -143,16 +143,6 @@
   set(HAVE_KF5Completion 1)
 endif()
 
-find_package(LibKWorkspace CONFIG)
-set_package_properties(LibKWorkspace
-  PROPERTIES
-  TYPE OPTIONAL
-  PURPOSE "Required for KTorrent's shutdown plugin"
-)
-if(LibKWorkspace_FOUND)
-  set(HAVE_LibKWorkspace 1)
-endif()
-
 #find_package(Qt${QT_MAJOR_VERSION} ${QT_MIN_VERSION} OPTIONAL_COMPONENTS Multimedia MultimediaWidgets)
 
 find_package(Taglib)
