From c5b182922af4608c36b32654302630ad4d6a6e7d Mon Sep 17 00:00:00 2001
From: Noah Davis <noahadvs@gmail.com>
Date: Tue, 9 Jul 2024 23:13:14 -0400
Subject: [PATCH] PanelView: fix crash in updateEnabledBorders when rootObject
 is null

BUG: 489940
---
 shell/panelview.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/shell/panelview.cpp b/shell/panelview.cpp
index 253ebcf63da..de6607f6029 100644
--- a/shell/panelview.cpp
+++ b/shell/panelview.cpp
@@ -1787,6 +1787,9 @@ bool PanelView::edgeActivated() const
 
 void PanelView::updateEnabledBorders()
 {
+    if (!rootObject()) {
+        return;
+    }
     KSvg::FrameSvg::EnabledBorders borders = KSvg::FrameSvg::AllBorders;
     if (m_backgroundHints == Plasma::Types::NoBackground) {
         borders = KSvg::FrameSvg::NoBorder;
-- 
GitLab

